kobo/apps/superuser_stats/tasks.py

```diff
@@ -79,6 +87,235 @@ def get_row_for_country(code_: str, label_: str):
+            date__range=(start_date, end_date),
+        ).aggregate(
+            Sum('counter')
+        )['counter__sum']
+        if domain_assets[domain] else 0
+        for domain in domain_users.keys()
+    }
+
+    # create the CSV file
+    columns = ['Email Domain', 'Users', 'Projects', 'Submissions']
+
+    default_storage = get_storage_class()()
+    with default_storage.open(output_filename, 'wb') as output:
+        writer = unicodecsv.writer(output)
+        writer.writerow(columns)
+
+        for domain, users in domain_users.most_common():
+            row = [
+                domain,
+                users,
+                domain_assets[domain],
+                domain_submissions[domain]
+            ]
+            writer.writerow(row)
+
+
+@shared_task
+def generate_forms_count_by_submission_range(output_filename):
+    # List of submissions count ranges
+    ranges = [
+        (0, 0),
+        (1, 500),
+        (501, 1000),
+        (1001, 10000),
+        (10001, 50000),
+        (50001, None)
+    ]
+
+    # store data for csv
+    data = []
+
+    today = datetime.datetime.today()
+    date = today - relativedelta(months=12)
+
+    for r in ranges:
+        forms_count = KobocatXForm.objects.filter(
+            date_created__gte=date,
+            num_of_submissions__range=r,
+        ).values('pk').count()
+        data.append([f'{r[0]}-{r[1]}', forms_count])
+
+    headers = ['Range', 'Count']
+
+    # Crate a csv with output filename
+    default_storage = get_storage_class()()
+    with default_storage.open(output_filename, 'wb') as output:
+        writer = unicodecsv.writer(output)
+        writer.writerow(headers)
+        writer.writerows(data)
+
+
+@shared_task
+def generate_media_storage_report(output_filename):
+    attachments = KobocatUserProfile.objects.all().values(
+        'user__username',
+        'attachment_storage_bytes',
+    )
+
+    data = []
+
+    for attachment_count in attachments:
+        data.append([
+            attachment_count['user__username'],
+            attachment_count['attachment_storage_bytes'],
+        ])
+
+    headers = ['Username', 'Storage Used (Bytes)']
+
+    default_storage = get_storage_class()()
+    with default_storage.open(output_filename, 'wb') as output:
+        writer = unicodecsv.writer(output)
+        writer.writerow(headers)
+        writer.writerows(data)
+
+
+@shared_task
+def generate_user_count_by_organization(output_filename: str):
+    # get users organizations
+    organizations = User.objects.filter(
+        extra_details__data__has_key='organization'
+    ).values_list(
+        'extra_details__data__organization', flat=True
+    ).distinct().order_by(
+        'extra_details__data__organization'
+    )
+
+    data = []
+
+    for organization in organizations:
+        count = User.objects.filter(
+            extra_details__data__organization=organization
+        ).count()
+        data.append([organization, count])
```

Comment:
I think, all of this could be simplified with 
```python 
organizations = User.objects.filter(
    extra_details__data__has_key='organization'
).values('extra_details__data__organization').annotate(total=Count('extra_details__data__organization'))

data = [[o['extra_details__data__organization'], o['total']] for o in organizations]

```